recipes/cminpack/all/conanfile.py

```diff
@@ -0,0 +1,116 @@
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools import files
+from conan import ConanFile
+import os
+
+required_conan_version = ">=1.45.0"
+
+
+class CMinpackConan(ConanFile):
+    name = "cminpack"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "About A C/C++ rewrite of the MINPACK software (originally in FORTRAN)" \
+                  "for solving nonlinear equations and nonlinear least squares problems"
+    topics = ("nonlinear", "solver")
+    homepage = "http://devernay.free.fr/hacks/cminpack/"
+    license = "LicenseRef-CopyrightMINPACK.txt"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_EXAMPLES"] = "OFF"
+        tc.variables["CMINPACK_LIB_INSTALL_DIR"] = "lib"
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
+        tc.generate()
+
+    def layout(self):
+        cmake_layout(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+        # cminpack is a c library
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+
+    def source(self):
+        files.get(self, **self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self.source_folder)
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        cmake = CMake(self)
+        cmake.install()
+
+        files.rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        files.copy(self, "CopyrightMINPACK.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        files.rmdir(self, os.path.join(self.package_folder, "share")) # contains cmake config files
+
+    def _library_postfix(self):
+        postfix = ""
+        if not self.options.shared:
+            postfix += "_s" # for static
+        if self.settings.build_type == "Debug":
+            postfix += "_d"
+
+        return postfix
+
+    def package_info(self):
+        minpack_include_dir = os.path.join("include", "cminpack-1")
+        
```

Comment:
```suggestion
        self.cpp_info.set_property("cmake_target_name", "cminpack")
```

The idea is transfer the default cmake target to `cminpack`, so, in case you link to the target `cminpack` it will link both libraries. As result, `cminpack::cminpack` should link only to the single library.

I checked locally, and it seems work: 

I enabled `CMAKE_VERBOSE_MAKEFILE` to see the output.

```
----Running------
> cmake --build "/home/conan/project/test_package/build/Release" '--' '-j5'
-----------------
/opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/bin/cmake -S/home/conan/project/test_package -B/home/conan/project/test_package/build/Release --check-build-system CMakeFiles/Makefile.cmake 0
/opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/bin/cmake -E cmake_progress_start /home/conan/project/test_package/build/Release/CMakeFiles /home/conan/project/test_package/build/Release//CMakeFiles/progress.marks
/usr/bin/make  -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/home/conan/project/test_package/build/Release'
/usr/bin/make  -f CMakeFiles/cminpack_test_float.dir/build.make CMakeFiles/cminpack_test_float.dir/depend
/usr/bin/make  -f CMakeFiles/cminpack_test_double.dir/build.make CMakeFiles/cminpack_test_double.dir/depend
make[2]: Entering directory '/home/conan/project/test_package/build/Release'
cd /home/conan/project/test_package/build/Release && /opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/bin/cmake -E cmake_depends "Unix Makefiles" /home/conan/project/test_package /home/conan/project/test_package /home/conan/project/test_package/build/Release /home/conan/project/test_package/build/Release /home/conan/project/test_package/build/Release/CMakeFiles/cminpack_test_double.dir/DependInfo.cmake --color=
make[2]: Entering directory '/home/conan/project/test_package/build/Release'
cd /home/conan/project/test_package/build/Release && /opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/bin/cmake -E cmake_depends "Unix Makefiles" /home/conan/project/test_package /home/conan/project/test_package /home/conan/project/test_package/build/Release /home/conan/project/test_package/build/Release /home/conan/project/test_package/build/Release/CMakeFiles/cminpack_test_float.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/conan/project/test_package/build/Release'
make[2]: Leaving directory '/home/conan/project/test_package/build/Release'
/usr/bin/make  -f CMakeFiles/cminpack_test_float.dir/build.make CMakeFiles/cminpack_test_float.dir/build
/usr/bin/make  -f CMakeFiles/cminpack_test_double.dir/build.make CMakeFiles/cminpack_test_double.dir/build
make[2]: Entering directory '/home/conan/project/test_package/build/Release'
make[2]: Entering directory '/home/conan/project/test_package/build/Release'
[ 50%] Linking C executable cminpack_test_double
[ 50%] Linking C executable cminpack_test_float
/opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/cminpack_test_float.dir/link.txt --verbose=1
/opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/cminpack_test_double.dir/link.txt --verbose=1
/usr/bin/gcc -m64 -O3 -DNDEBUG -m64 CMakeFiles/cminpack_test_double.dir/tchkderc.c.o -o cminpack_test_double  /home/conan/.conan/data/cminpack/1.3.8/_/_/package/19729b9559f3ae196cad45cb2b97468ccb75dcd1/lib/libcminpack_s.a -lm 
/usr/bin/gcc -m64 -O3 -DNDEBUG -m64 CMakeFiles/cminpack_test_float.dir/tchkderc.c.o -o cminpack_test_float  /home/conan/.conan/data/cminpack/1.3.8/_/_/package/19729b9559f3ae196cad45cb2b97468ccb75dcd1/lib/libcminpacks_s.a -lm 
make[2]: Leaving directory '/home/conan/project/test_package/build/Release'
make[2]: Leaving directory '/home/conan/project/test_package/build/Release'
[100%] Built target cminpack_test_float
[100%] Built target cminpack_test_double
make[1]: Leaving directory '/home/conan/project/test_package/build/Release'
/opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/bin/cmake -E cmake_progress_start /home/conan/project/test_package/build/Release/CMakeFiles 0
```

Reply:
Wow... it really seems to work. But I still don't understand why!

Unfortunately, I could not really follow your explanation. 

> ...  in case you link to the target cminpack it will link both libraries.

??? Do you mean cminpack without namespace prefix? Ideally, this target is deleted, but hey.... we probably cannot have everything 😏 

Awesome job!

Reply:
>  Do you mean cminpack without namespace prefix

Yes! But you can use another target name, if you want.